cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -440,19 +440,15 @@ public void stop(Id.Program programId) throws Exception {
 
   private void stopProgramIfRunning(Id.Program programId)
     throws InterruptedException, ExecutionException {
-    ProgramRuntimeService.RuntimeInfo programRunInfo = findRuntimeInfo(programId.getNamespaceId(),
-                                                                       programId.getApplicationId(),
-                                                                       programId.getId(),
-                                                                       programId.getType(),
-                                                                       runtimeService);
+    ProgramRuntimeService.RuntimeInfo programRunInfo = findRuntimeInfo(programId, runtimeService);
     if (programRunInfo != null) {
-      doStop(programRunInfo);
+      doStop(programRunInfo, programId);
     }
   }
 
-  private void doStop(ProgramRuntimeService.RuntimeInfo runtimeInfo)
-    throws ExecutionException, InterruptedException {
-    Preconditions.checkNotNull(runtimeInfo, UserMessages.getMessage(UserErrors.RUNTIME_INFO_NOT_FOUND));
+  private void doStop(ProgramRuntimeService.RuntimeInfo runtimeInfo,
+                      Id.Program programId) throws ExecutionException, InterruptedException {
+    Preconditions.checkNotNull(runtimeInfo, UserMessages.getMessage(UserErrors.RUNTIME_INFO_NOT_FOUND), programId);
```

Comment:
The calling code was always passing a non-null runtimeInfo, the error message (and the HTTP code for that case) was never happening. I simplified the code based on that.